src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -89,7 +89,7 @@ impl Step for Std {
         let stage = self.custom_stage.unwrap_or(builder.top_stage);
 
         let target = self.target;
-        let compiler = builder.compiler(stage, builder.config.build);
+        let compiler = builder.compiler(stage, builder.config.host_target);
```

Comment:
Discussion [NOMENCLATURE 2/2]: we'll need to be a bit careful here, we can have:

1. "Usual" cross-compile scenario: in this scenario, we have **build platform** == **host platform**, but both are different from the **target** platform.
2. Canadian-cross scenario: in this scenario, we have **build** platform != **host** platform != **target** platform.

In bootstrap config terms, this is like

```
build = "x86_64-pc-windows-msvc"
hosts = ["x86_64-unknown-linux-gnu"]
targets = ["wasm32-unknown-unknown"]
```

And apparently this kind of configuration is indeed used by people (I think most often distro maintainers?).

Reply:
Well, this is an unfortunate naming on bootstrap's part. What bootstrap calls build is the literal host, i.e. the computer that runs bootstrap (unless you're cross-compiling bootstrap itself, in which case.. good luck). What bootstrap calls `hosts` are targets for which we build a compiler and what bootstrap calls `targets` are targets for which we build stdlib :upside_down_face: